network/peer.go

```diff
@@ -836,3 +868,50 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time().Unix(),
+	})
+}
+
+// assumes stateLock is held
+func (p *peer) releaseAliases() {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	for _, alias := range p.aliases {
+		str := alias.ip.String()
+		delete(p.net.peerAliasIPs, str)
+
+		p.net.log.Verbo("released alias %s for peer %s", str, p.id.String())
+	}
+	p.aliases = p.aliases[:0]
```

Comment:
I think we can do: `p.aliases = nil` here so the slice can get garbage collected

Reply:
Yep...for some reason I was thinking it would make sense to preserve the array memory but doesn't make sense given we are destroying the peer record 🤦